Some changes and fixes to the transformer tester

cvalarida opened the pull request:
## Description
Made some changes and fixes to the transformer tester. Kinda like the title says. :sweat_smile: 

- Default `--transform-unpublished` to true, which seemed to make sense since we're using this to test transformers
- Actually made `--transform-unpublished` _do_ something (was broken before)
- Added a `--buildtype` flag for when we want to test the transformations against another buildtype.

Commit:
Destructure the correct property :facepalm:

Commit:
Since we're using this to test, default to true

Commit:
Add buildtype arg

theodur commented on script/cms/test-bundle-transformers.js:
```diff
@@ -66,9 +65,16 @@ const {
 
+const contentDir = path.resolve(
+  __dirname,
+  `../../.cache/${buildtype}/cms-export-content`,
```
Moving forward does this mean the `buildtype` flag is required? I noticed when I exclude it from the command line I get an error because it's `undefined`.

theodur commented on script/cms/test-bundle-transformers.js:
```diff
@@ -66,9 +65,16 @@ const {
   bundle,
   help,
-  transformUnpublished,
+  'transform-unpublished': transformUnpublished,
```
After testing it doesn't seem like the default value for `transform-unpublished` is being set to true.

Commit:
Uh, so it should be defaultValue...

cvalarida commented:
Ahh, good catch on both those, @theodur! Turns out it should be [`defaultValue`](https://github.com/75lb/command-line-args/blob/master/doc/option-definition.md#module_option-definition--OptionDefinition+defaultValue), not `default`. :man_facepalming:

theodur commented:
`defaultValue` fixed it! In the future will we need a way to set `transform-unpublished` to false? Currently you can't set it false through the flag in the terminal. If not, and it should always be set to true then maybe we won't need the `transform-unpublished` flag anymore.

cvalarida commented:
:thinking: I guess I assumed `--transform-unpublished false` would do the trick, but I never did test that. Good point. I could change it to `--transform-only-published` and have it default to `false`. That would accomplish the same thing.

Commit:
Switch the flag to explicitly transform only published

Commit:
Filter untransformed entities

Because if we don't transform unpublished entities, `assembleEntityTree`
will return `null` and our count will be off.

cvalarida commented:
Also updated the transformed entities count at the end to—ahem—_account_ for not transforming unpublished entities.
![image](https://user-images.githubusercontent.com/12970166/88864315-c1df1900-d1b9-11ea-8878-41de7a26985e.png)
...
![image](https://user-images.githubusercontent.com/12970166/88864324-cb688100-d1b9-11ea-8f1d-7d1fac8da152.png)
![image](https://user-images.githubusercontent.com/12970166/88864336-d15e6200-d1b9-11ea-88d8-7fb3fa1f6ae3.png)

Commit:
Merge branch 'master' into transformer-tester-unpublished-fix-cv

theodur reviewed (approved):
I like the `transform-only-published` flag! And I forgot about unpublished entities being counted in the number of entities transformed, nice call. These changes look good!
